include/cuco/static_set_ref.cuh

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/equal_wrapper.cuh>
+#include <cuco/function.hpp>
+#include <cuco/sentinel.cuh>  // TODO .hpp
+
+#include <cuda/std/atomic>
+
+namespace cuco {
+namespace experimental {
+
+/**
+ * @brief Device reference of static_set.
+ */
+template <typename Key,
+          cuda::thread_scope Scope,
+          typename KeyEqual,
+          typename ProbingScheme,
+          typename StorageRef,
+          typename... Functions>
```

Comment:
I don't know why, but "Operators" feels like a more appropriate name. "Functions" feels too generic like it can be any function.

Reply:
I went with Operators for like half of all my prototypes then somehow switched to Functions. The naming struggle is real 😅

Wikipedia says:
> In mathematics, an operator is generally a mapping or function that acts on elements of a space to produce elements of another space (possibly the same space, sometimes required to be the same space). There is no general definition of an operator, but the term is often used in place of function when the domain is a set of functions or other structured objects.

This sounds to me like Operators is the right choice.

Reply:
Either `Operators` or `Operations` sounds good to me.